addons/bus/models/ir_websocket.py

```diff
@@ -82,54 +40,40 @@ def _prepare_subscribe_data(self, channels, last):
         last = 0 if last > self.env["bus.bus"].sudo()._bus_last_id() else last
-        str_presence_channels = {
-            c for c in channels if isinstance(c, str) and c.startswith("odoo-presence-")
-        }
-        presence_channels = self._build_presence_channel_list(
-            [tuple(c.replace("odoo-presence-", "").split("_")) for c in str_presence_channels]
-        )
-        # There is a gap between a subscription client side (which is debounced)
-        # and the actual subcription thus presences can be missed. Send a
-        # notification to avoid missing presences during a subscription.
-        domain = expression.AND(
-            [
-                [("last_poll", ">", datetime.now() - timedelta(seconds=2))],
-                expression.OR(self._get_missed_presences_identity_domains(presence_channels)),
-            ]
-        )
-        # sudo: bus.presence: can access presences linked to presence channels.
-        missed_presences = self.env["bus.presence"].sudo().search(domain)
-        all_channels = OrderedSet(presence_channels)
-        all_channels.update(
-            self._build_bus_channel_list([c for c in channels if c not in str_presence_channels])
-        )
-        return {"channels": all_channels, "last": last, "missed_presences": missed_presences}
+        return {"channels": channels, "last": last}
 
     def _subscribe(self, og_data):
         data = self._prepare_subscribe_data(og_data["channels"], og_data["last"])
         dispatch.subscribe(data["channels"], data["last"], self.env.registry.db_name, wsrequest.ws)
-        if bus_target := self._get_missed_presences_bus_target():
-            data["missed_presences"]._send_presence(bus_target=bus_target)
 
     def _update_bus_presence(self, inactivity_period, im_status_ids_by_model):
-        if self.env.user and not self.env.user._is_public():
-            self.env['bus.presence'].update_presence(
-                inactivity_period,
-                identity_field='user_id',
-                identity_value=self.env.uid
-            )
+        """
+            Update the presence information of users on the bus.
+
+            - This method is hardcoded to be one of the few methods that can be called via the websocket
+            route due to Odoo's architecture and its use of proxying on odoo.sh.
+            - The restriction ensures that only essential methods are exposed to save server resources
+            and avoid unnecessary polling.
+
+            :param int inactivity_period: The period (in milliseconds) since last user activity.
+            :param dict im_status_ids_by_model: A dictionary containing IM status IDs referenced by model.
+        """
 
     def _on_websocket_closed(self, cookies):
-        if self.env.user and not self.env.user._is_public():
-            self.env["bus.presence"].search([("user_id", "=", self.env.uid)]).status = "offline"
+        """
+            Handle the closure of a websocket connection.
```

Comment:
Could be shorter. `Function invoked upon WebSocket termination`. What is done in the overrides does not matter.